chiselFrontend/src/main/scala/chisel3/core/Data.scala

```diff
@@ -430,3 +435,34 @@ object WireInit {
     x
   }
 }
+
+/** RHS (source) for Invalidate API.
+  * Causes connection logic to emit a DefInvalid when connected to an output port (or wire).
+  */
+object DontCare extends Element(width = UnknownWidth()) {
+  binding = DontCareBinding()
+  direction = ActualDirection.Output
+  override def cloneType = DontCare
+
+  // DontCare may only be connected to Ports and Wires (and Unbound), and only as a source
+  // Attempts to change its binding generate errors.
+  private[chisel3] override def bind(target: Binding, parentDirection: SpecifiedDirection) {
+    target match {
+      case _: DontCareBinding =>
+      case _ => Builder.error(s"DontCare binding may not be changed (to ${target.toString}).")
+    }
+  }
+
+  def toPrintable: Printable = PString("DONTCARE")
+
+  private[core] def connectFromBits(that: chisel3.core.Bits)(implicit sourceInfo:  SourceInfo, compileOptions: CompileOptions): Unit = {
+    Builder.error("connectFromBits: DontCare cannot be a connection sink (LHS)")
+  }
+
+  def do_asUInt(implicit sourceInfo: chisel3.internal.sourceinfo.SourceInfo, compileOptions: CompileOptions): chisel3.core.UInt = {
+    Builder.error("DontCare does not have a UInt representation")
+    0.U
+  }
+  // DontCare's match any type.
+  private[core] def typeEquivalent(that: chisel3.core.Data): Boolean = true
```

Comment:
Why?
I think it makes more sense for DontCares to not be equivalent to anything but another DoneCare, or to have this crash.

Reply:
Done.